### PR TITLE
18273 Add show-current to DatePicker

### PR DESCRIPTION
**Pull request by leodube-aot**
*Issue #:* /bcgov/entity#18273

*Description of changes:*
I needed specific functionality to improve the DatePicker for #18273. Currently, when we add a `min` restriction on the DatePicker in the Agm Extension Request component, the `min` valid date can be several months, or even over a year into the future. When the picker loads, is does not default to the earliest valid date, and instead shows the current date.

Screenshot 2023-11-20 at 1 17 47 PM

Passing the minimum date to `showCurrent` will now display the earliest valid date by default when the component loads.

```
<DatePicker
  ...
  initialValue="2025-06-06"   // initialValue also needs to be provided for the title date button to update as well
  minDate="2025-06-06"
  showCurrent="2025-06-06"
  ...
/>
```

Screenshot 2023-11-20 at 1 40 50 PM

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: docs/.nojekyll
Comment (severinbeauvais) [inline docs/.nojekyll:1]: Don't delete this file. See README.

File: src/components/date-picker/DatePicker.vue [hunk 1]
```diff
@@ -96,6 +97,7 @@ export default class DatePicker extends Mixins(DateMixin) {
   @Prop({ default: () => [] }) readonly inputRules!: Array<(v) => boolean | string>
   @Prop({ default: false }) readonly disablePicker!: boolean
   @Prop({ default: '' }) readonly initialValue!: string
+  @Prop({ default: true }) readonly showCurrent!: boolean | string
```
Comment (severinbeauvais) [inline src/components/date-picker/DatePicker.vue:100]: Does the default value maintain the current behaviour?

Comment (leodube-aot) [reply src/components/date-picker/DatePicker.vue:100]: Yes, the default value on `v-date-picker` for it is `true`: https://v2.vuetifyjs.com/en/api/v-date-picker/#props

Comment (severinbeauvais) [reply src/components/date-picker/DatePicker.vue:100]: Perfect answer, thanks! 😃

**Review by severinbeauvais — APPROVED** [review]

File: docs/.nojekyll
Comment (severinbeauvais) [reply docs/.nojekyll:1]: LGTM

**Review by ketaki-deodhar — APPROVED** [review]
Looks good!

**Review by JazzarKarim — APPROVED** [review]
👍

**Pull request merged**